tfs/src/main/java/hudson/plugins/tfs/JenkinsEventNotifier.java

```diff
@@ -81,17 +82,22 @@ public static String getApiJson(final String url) {
             request.addHeader("User-Agent", "Jenkins-Self");
 
             final HttpResponse response = client.execute(request);
-
```

Comment:
@jpricketMSFT This code shouldn't attempt to get the full json over an http connection, as @smile21prc notes.
- It's very expensive
- It won't work with authentication (easily

The required fields should be picked from the run, serialized into json, and then sent off.

Reply:
Unfortunately, the json can be added to by plugins and there is no set list of properties that event listeners need. I will try to find another solution, but it will take some time.

Reply:
In this case, I prefer not including the overall fix in this PR, as it's not blocking my feature. I'll leave the long-term fix to Jason.